### DISCUSSION TITLE
golangci-lint-langserver not working with workspace diagnostics

### DISCUSSION THREAD

**Discussion by ClaraSmyth** [helix-editor/helix#9729] [Troubleshooting]
### Summary

When using golangci-lint-langserver it works fine within the open buffers but when using the workspace diagnostic picker it will not show diagnostics for files not currently open.

### Reproduction Steps

 

I tried this:

1. `hx .`
2. `select any file in project`
3. `space + D`

I expected this to happen:
To display diagnostics across all files in the workspace where the go.mod file is at the root

Instead, this happened:
Only diagnostics for the open buffers are displayed

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-02-26T16:06:50.277 helix_lsp::transport [ERROR] golangci-lint-lsp err <- "golangci-lint-langserver: connections opened\n"
2024-02-26T16:06:52.636 helix_lsp::transport [ERROR] golangci-lint-lsp err <- "2024/02/26 16:06:52 jsonrpc2 handler: notification \"exit\" handling error: jsonrpc2: code -32601 message: method not supported: exit\n"
2024-02-26T16:06:52.638 helix_lsp::transport [ERROR] golangci-lint-lsp err: <- StreamClosed
2024-02-26T16:06:52.638 helix_lsp::transport [ERROR] err: <- Other(failed to send a message to server

Caused by:
    channel closed)
2024-02-26T16:07:31.003 helix_lsp::transport [ERROR] golangci-lint-lsp err <- "golangci-lint-langserver: connections opened\n"

```

</details>

### Platform

Linux

### Terminal Emulator

wezterm 20240211-083735-b61c49fa

### Installation Method

source

### Helix Version

23.10-332-gcd02976f

**Comment by cbrake** [comment]
do you mind sharing your language config on how you enable this?

**Comment by gringonivoli** [comment]
the same thing happens to me in a python project. Python LSP default config. 

**Platform:** macOS Sonoma 14.3.1
**Installation Method:** brew
**Helix version:** helix 23.10 (f6021dd0)

**Comment by cbrake** [comment]
This is how I got golantci-lint-langserver working on the Arch Linux packaged version of Helix:

https://community.tmpdir.org/t/helix-editor/1244/9?u=cbrake

@ClaraSmyth Regarding the original question, LazyVIM behaves the same as Helix and only shows me workspace diagnostics for the open buffers:

![image](https://github.com/helix-editor/helix/assets/402813/9dfe507e-2768-4587-98f4-e4e48508896d)

(there are many more problems in this codebase).

So, I suspect this is working as intended.

You could probably open all source files or something ...

**Reply by ClaraSmyth** [reply]
Hi, thanks for the reply. I did experiment with a few different flags to try get it working how I expect but couldn't find any solution. So i just left my language config as the default for now.

I might try to get a working solution using efm-langserver instead.

**Reply by ClaraSmyth** [reply]
I found what seems to be a working solution using efm-langserver. Gives me what I was looking for at least.

Heres a link to my language config if you happen to be interested - https://github.com/ClaraSmyth/dotfiles/blob/main/.config/helix/languages.toml#L24
